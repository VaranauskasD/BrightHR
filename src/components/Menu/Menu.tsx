import React, { ReactNode, ChangeEvent } from 'react'
import styled from 'styled-components'

import { BsFillArrowLeftCircleFill } from 'react-icons/Bs'

import { Filter } from '../Filter'

export interface MenuProps {
  handleChange: (event: ChangeEvent<HTMLInputElement>) => void
}

const MenuWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const StyledMenu = styled.div`
  display: flex;
  justify-content: space-between;
  border-radius: 20px 20px 0 0;
  padding: 8px;
  height: 48px;
  background: ${(props) => `${props.theme.colors.primary}`};
`

const FilterWrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: center;
`

const Navigation = styled.div`
  display: flex;
  flex: 1;
`

const StyledButton = styled.button`
  display: flex;
  align-items: center;
  background: transparent;
  border: none;
  outline: none;
  cursor: pointer;
  :focus-within,
  :hover {
    box-shadow: 0 0 4px rgba(0, 0, 0, 0.5);
  }
`

const StyledArrowIcon = styled(BsFillArrowLeftCircleFill)`
  min-height: 24px;
  min-width: 24px;
  color: ${(props) => `${props.theme.colors.secondary}`};
`

export const Menu = (props: MenuProps) => {
  return (
    <MenuWrapper>
      <StyledMenu>
        <StyledButton type="button" aria-label="Back">
          <StyledArrowIcon />
        </StyledButton>
        <FilterWrapper>
          <Filter handleChange={props.handleChange} />
        </FilterWrapper>
      </StyledMenu>
    </MenuWrapper>
  )
}

export default Menu
