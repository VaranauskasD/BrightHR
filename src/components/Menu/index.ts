export type { MenuProps } from './Menu'
export { Menu } from './Menu'
