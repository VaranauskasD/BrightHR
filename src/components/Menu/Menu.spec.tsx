import { customRender as render } from '../../utils/test'
import { Menu, MenuProps } from './Menu'

let mockMenuProps: MenuProps
describe('Menu', () => {
  beforeEach(() => {
    mockMenuProps = {
      handleChange: () => {},
    }
  })

  it('renders correctly', async () => {
    const { asFragment } = render(<Menu {...mockMenuProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
