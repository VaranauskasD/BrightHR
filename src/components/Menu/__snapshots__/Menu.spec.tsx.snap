// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-eCYdqJ bcsQqv"
  >
    <div
      class="sc-jSMfEi gZdlnv"
    >
      <div
        class="sc-gKXOVf gtvVWl"
      >
        <div
          class="sc-bczRLJ bxjVfh"
        >
          <label
            class="sc-gsnTZi fDnEkf"
          >
            Filter files
          </label>
          <svg
            aria-hidden="true"
            class="sc-dkzDqf dMgMsg"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
            />
          </svg>
          <input
            class="sc-hKMtZM lchXNC"
            data-testid="filter"
            placeholder="Filter"
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
